tests/unit/basic_crawler/test_basic_crawler.py

```diff
@@ -671,6 +679,20 @@ async def handler(context: BasicCrawlingContext) -> None:
     assert (await store.get_value('foo')) == 'bar'
 
 
+async def test_context_use_state(mock_event_manager: EventManager) -> None:
+    crawler = BasicCrawler()
+
+    @crawler.router.default_handler
+    async def handler(context: BasicCrawlingContext) -> None:
+        await context.use_state('state', {'hello': 'world'})
+        await asyncio.sleep(0.1)
```

Comment:
Could the sleep go after `crawler.run` instead? I assume it's to ensure key-value store consistency.

Reply:
Okay, would that mean that if I did this in production, state changes in the last request handler run might not be saved?

Reply:
That's right.

I tested this with different `persist_state_interval` values for `EventManager`, without forced sleep the saves only occur at values less than 10 milliseconds.

So if we have to ensure that the state is persisted, all `Event.PERSIST_STATE` listeners should be forced run when the `EventManager` is closed.